src/app/icd/ICDManager.cpp

```diff
@@ -152,6 +161,10 @@ void ICDManager::UpdateOperationState(OperationalState state)
             {
                 ChipLogError(AppServer, "Failed to set Polling Interval: err %" CHIP_ERROR_FORMAT, err.Format());
             }
+
+#if ICD_REPORT_ON_ENTER_ACTIVE_MODE
+            mStateObserver->OnEnterActiveMode();
+#endif
```

Comment:
Why does this part need to be ifdefed?  Seems like the real ifdefing should be in the report scheduler; we just notify our observer, and what it does is up to it.